NetKet/GroundState/pyground_state.hpp

```diff
@@ -31,33 +31,77 @@ namespace netket {
 
 void AddGroundStateModule(py::module &m) {
   auto subm = m.def_submodule("gs");
```

Comment:
I think that `netket.gs.vmc.Vmc` is too deep of a module hierarchy, we should try to simplify it. 
`netket.Vmc` doesn't look bad to me, but let's discuss more

Reply:
I think it would make sense to separate the VMC code from the ED code. So `netket.vmc` and `netket.ed` seem fine to me.

Reply:
Although strictly speaking `ed` is a bit of a misnomer for exact time evolution.

Reply:
Yes, `netket.exact` sounds good.

Regarding `netket.vmc`: I see your point, however, we might get `netket.TdVmc` in the future. (In principle, we could also call it `netket.vmc.TimeEvolution` or similar.) Also, there is `netket.vmc.Iterator` (which users would usually not see, but has to be somewhere).